google-cloud-storage/lib/google/cloud/storage/file/signer.rb

```diff
@@ -42,10 +42,7 @@ def self.from_bucket bucket, path
           ##
           # The external path to the file.
           def ext_path
-            escaped_path = String(@path).split("/").map do |node|
-              CGI.escape node
-            end.join("/")
-            "/#{CGI.escape @bucket}/#{escaped_path}"
+            Addressable::URI.escape "/#{@bucket}/#{@path}"
           end
```

Comment:
@frankyn I think we are relying here on the behavior (seen in the unit tests below) of `Addressable::URI.encode` to not encode forward slashes at all.

Reply:
ah, okay, I take back my last comment about not removing the split. Thanks for clarifying @quartzmo.